CHANGES.md

```diff
@@ -1,6 +1,462 @@
+* added an error for exported routines that have `where` clauses
+* improved error messages for generic first-class functions
+* added a warning for certain races that occur when promoting a method
+* upgraded the format of internal error messages
+
+Execution-time Checks
+---------------------
+* added an error when remote records are passed to external procedures
+
+Bug Fixes
+---------
+* fixed support for task-private array variables
+* fixed bugs in `writef()`/`readf()`/`format()` for enums without int values
+* fixed bugs so that `LinearAlgebra` routines can accept rank-change slices
+* made empty `Spawn.spawn()` calls throw errors instead of segfaulting
+* fixed a bug when running `subprocess.wait` more than once
+* resolved problems when multiple extern blocks `#include`d the same header
+* resolved problems for extern blocks with mutually recursive `struct`s
+* made `mason search` warn about bad manifest filenames instead of halting
+* fixed a bug in which `mason doc` expected project dirs to match package names
+* fixed a bug in which certain `try!` patterns caused internal errors
+* fixed a bug in which I/O couldn't read integers starting with `+` like `+10`
+* iterating over tuples now yields `ref` or `const ref` elements
+* `on s` for `owned` or `shared` objects now runs where the object is allocated
+* fixed support for forall loops over zippered loop expressions
+* fixed support for forall loops over tuple expansion expressions
+* enabled tuples to contain `owned` objects
+* fixed a bug in range.member(range)
+* fixed some bugs/inconsistencies in methods and functions on ranges
+* fixed support for `cstdlib` atomics for clang and llvm compilers
+* fixed some issues with `--library*` compilation using the LLVM back-end
+  (see https://chapel-lang.org/docs/1.19/technotes/libraries.html#llvm))
+* fixed `setchplenv` scripts for `pyenv` users
+* fixed a bug in which `PATH` was not quoted in `chpldoc` & `chplvenv` builds
+* fixed a problem where `setchplenv` added `' '` to MANPATH
+* fixed a bug in which comparing rank-variant domains generated a compiler error
+* fixed a bug in which global extern consts were incorrectly localized
+* fixed a problem with override-checking for functions with `where`-clauses
+* fixed a problem with control flow analysis in functions with `catch` blocks
+* converted memcpy() calls that may overlap to use memmove()
+* fixed some incorrect buffer sizes found using gcc 8.2 static analysis
+* fixed target-compiler optimization flags for non-x86 architectures
+* fixed problems caused by toggling `CHPL_LLVM` while running `make`
+* fixed an error in the documentation of `string.match`
+
+Third-Party Software Changes
+----------------------------
+* upgraded GASNet to GASNet-EX version 2018.9.0
+* upgraded hwloc to version 1.11.11
+* upgraded qthreads to version 1.14
+* upgraded re2 to commit 0a6326b
+* updated the bundled version of LLVM to 7.0.1 and began storing it unpacked
```

Comment:
Maybe change this to "upgraded llvm to version 7.0.1" to match the others (the doc references here also seem unnecessary)

Reply:
I have no objection to changing the wording.  Let's make sure to keep the part about it being unpacked because that is significant to the way it is accessed.